### PR TITLE
Implement Request POST Methods

### LINKED ISSUE #445: BE - API Methods Implementation for Requests
### Issue Description

Following POST, PUT API methods should be implemented for workspaces.

### Steps

- [x] #446 
- [x] #447 
- [x] #448 
- [x] #449 

### Deadline for the Issue

17.11.2023 - 23:59

### PR DESCRIPTION
**Pull request by Bera0422**
Implemented POST and PUT methods for collaboration and review requests to send and update the status. Usage examples are provided in Postman. This PR fixes #445

### CODE REVIEW

**Comment by hakanaktas0** [comment]
Why do ReviewRequest and CollabRequest APIs require title and body fields in the request body? 
Other than this, it looks OK.

File: project/backend/api/views.py [hunk 1]
```diff
@@ -346,3 +346,39 @@ def get_workspace_from_id(request):
...
+    serializer = CollaborationRequestSerializer(data=request.data)
+    if serializer.is_valid():
+        serializer.save()
+        return Response(serializer.data, status=201)
+    return Response(serializer.errors, status=400)
+
+@api_view(['PUT'])
+def update_request_status(request):
+    try:
+        req = Request.objects.get(pk=request.data.get('workspace_id'))
```
Comment (hakanaktas0) [inline project/backend/api/views.py:362]: workspace_id?

**Review by hakanaktas0 — CHANGES_REQUESTED** [review]

**Comment by Bera0422** [comment]
> Why do ReviewRequest and CollabRequest APIs require title and body fields in the request body? Other than this, it looks OK.

I thought we need them to create and send requests, don't we?

File: project/backend/api/views.py (hunk 1 above)
Comment (Bera0422) [reply project/backend/api/views.py:362]: Workspace model uses "workspace_id" as its primary key. Should we access the workspaces in different way?

**Review by hakanaktas0 — APPROVED** [review]
All looks good.

**Pull request merged**